ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixParticipant.java

```diff
@@ -136,6 +138,40 @@ public boolean setReplicaSealedState(ReplicaId replicaId, boolean isSealed) {
     }
   }
 
+  @Override
+  public boolean setReplicaStoppedState(List<ReplicaId> replicaIds, boolean isStopped) {
+    Set<String> replicasToUpdate = new HashSet<>();
+    for (ReplicaId replicaId : replicaIds) {
+      if (!(replicaId instanceof AmbryReplica)) {
+        throw new IllegalArgumentException(
+            "HelixParticipant only works with the AmbryReplica implementation of ReplicaId");
+      }
+      replicasToUpdate.add(replicaId.getPartitionId().toPathString());
+    }
+    boolean setResult;
+    synchronized (helixAdministrationLock) {
+      logger.trace("Getting stopped replicas from instanceConfig");
+      List<String> stoppedList = getStoppedReplicas();
+      if (stoppedList == null) {
+        stoppedList = new ArrayList<>();
+      }
+      Set<String> stoppedReplicas = new HashSet<>(stoppedList);
+      if (!isStopped) {
+        replicasToUpdate.retainAll(stoppedReplicas);
+        logger.info("Removing replicas {} from stopped list", Arrays.toString(replicasToUpdate.toArray()));
```

Comment:
I think it might be useful to check whether `replicasToUpdate` is non zero size here so as to avoid "updates" that don't change anything.

Reply:
Also need a test that verifies that

Reply:
Same for the other way round i.e stopping (don't do the operation if nothing changes)

Reply:
yeah. good point! I will take your advice and make changes as well as add unit tests.